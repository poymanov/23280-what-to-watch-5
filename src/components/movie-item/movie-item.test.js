import React from "react";
import renderer from "react-test-renderer";
import MovieItem from "./movie-item";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";

const noop = () => {};

const movie = {
  id: 1,
  name: `test`,
  posterImage: `test`,
  previewImage: `test`,
  backgroundImage: `test`,
  backgroundColor: `test`,
  videoLink: `test`,
  previewVideoLink: `test`,
  description: `test`,
  rating: 1,
  scoresCount: 1,
  director: `test`,
  starring: [`test`],
  runTime: 1,
  genre: `test`,
  released: 1,
  isFavorite: false,
};

describe(`Should MovieItem render correctly`, () => {
  it(`Without trailer`, () => {
    const tree = renderer
      .create(<BrowserRouter history={browserHistory}>
        <MovieItem movie={movie} onMovieHover={noop} onMovieLeave={noop} isShowTrailer={false} />
      </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With trailer`, () => {
    const tree = renderer
      .create(<BrowserRouter history={browserHistory}>
        <MovieItem movie={movie} onMovieHover={noop} onMovieLeave={noop} isShowTrailer={true} />
      </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
