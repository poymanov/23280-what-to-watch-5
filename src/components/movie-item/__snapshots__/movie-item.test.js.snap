// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieItem render correctly With trailer 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <video
    autoPlay={true}
    height="175"
    muted={true}
    poster="test"
    src="test"
    width="336"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      test
    </a>
  </h3>
</article>
`;

exports[`Should MovieItem render correctly Without trailer 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="test"
      height="175"
      src="test"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      test
    </a>
  </h3>
</article>
`;
